### DISCUSSION TITLE
Moon phase by observation

### DISCUSSION THREAD

**Discussion by JiriSchmitt** [cosinekitty/astronomy#161] [Q&A] [answered]
Hello,

thank you for great engine. I have question how can I compute moon phase for observation. There is only possible by time, but it returns different result then for example on this page: https://www.timeanddate.com/moon/phases/

I can see hee for now 84.1% but your engine returns 227%. I use following code to compute percents (need percents between full moon and new moon): `360 - 227 = 133 / 180 * 100 = 73.88%`

I´m not sure if it is correct computing or how can I compute this correctly.

I also tried compute it by time from new moon:
```c#
var hours = (time - newMoon).TotalHours;
return (hours / 708.734136) * (360 / 100) * 100;
```
But also it is not correct. I gets 196.77 which is 90.68% by my conversion.

Could you help me please?
Thank you Jiri

**Comment by cosinekitty** [comment, marked as answer]
Hi @JiriSchmitt , and welcome to the project!  That is a great question. I realize the phrase "moon phase" is ambiguous. It looks like you are using C# (because I see TotalHours).  There are two different things "moon phase" can mean:

1. Relative ecliptic longitude. This is what the `MoonPhase` function is calculating. This is the traditional way of defining new moon, first quarter, full moon, and third quarter. It always goes from 0 to 360 on every orbit. In this method, we project the Moon's 3D position onto the ecliptic plane (the plane of the Earth's orbit around the Sun) and ignore how far above or below that plane the Moon is. Then we take the angle of that "shadow" of the Moon around its orbit, using the Sun's position as a reference.  So 0 = New Moon, 180 = Full Moon, etc.
2. Illumination fraction. This is what I believe you are looking for. This is the percentage of the Moon's face that appears to be illuminated by sunlight, as seen from the Earth. When the Moon is full, the illuminated fraction will be close to 100%, but not quite. There is a separate function called [`Illumination`](https://github.com/cosinekitty/astronomy/tree/master/source/csharp#Astronomy.Illumination) that does this calculation. It returns an [`IllumInfo`](https://github.com/cosinekitty/astronomy/tree/master/source/csharp#IllumInfo) data structure that contains several interesting values, among which is `phase_fraction`, a number between 0 and 1. Just multiply it by 100 to get a percentage.

I have updated the all of the "moonphase" demos, including the [C# moonphase demo](https://github.com/cosinekitty/astronomy/blob/master/demo/csharp/moonphase/moonphase.cs) to calculate both the ecliptic phase and the illumination fraction.  The key part is this excerpt:

```csharp
    IllumInfo illum = Astronomy.Illumination(Body.Moon, time);
    Console.WriteLine("{0} : Moon's illuminated fraction = {1:F2}%.", time, 100.0 * illum.phase_fraction);
```

I just ran the new version of `moonphase.cs` and here is its output:

```
$ dotnet run
2022-02-20T21:44:38.196Z : Moon's ecliptic phase angle = 231.206 degrees.
2022-02-20T21:44:38.196Z : Moon's illuminated fraction = 81.37%.

The next 10 lunar quarters are:
2022-02-23T22:33:00.612Z : Third Quarter
2022-03-02T17:35:25.324Z : New Moon
2022-03-10T10:46:08.996Z : First Quarter
2022-03-18T07:18:12.780Z : Full Moon
2022-03-25T05:37:50.281Z : Third Quarter
2022-04-01T06:25:02.198Z : New Moon
2022-04-09T06:48:20.907Z : First Quarter
2022-04-16T18:55:35.902Z : Full Moon
2022-04-23T11:56:56.727Z : Third Quarter
2022-04-30T20:28:46.178Z : New Moon
```

The timeanddate.com page you sent reports the current illuminated fraction is 81.4%, which rounds to the same value.

I hope this helps! Please let me know if you have any more questions.

**Reply by JiriSchmitt** [reply]
Hi @cosinekitty,

thank you very much it is exactly what I'm looking for.

And yes, I'm working in c# with your astronomy engine. I'm using it for small application for calculating biblical new moons and shabbat days which are calculated from new moon.

Best regards
